test/dummies/vars.go

```diff
@@ -0,0 +1,254 @@
+package dummies
+
+import (
+	csapi "github.com/apache/cloudstack-go/v2/cloudstack"
+	capcv1 "github.com/aws/cluster-api-provider-cloudstack/api/v1beta1"
+	"github.com/aws/cluster-api-provider-cloudstack/pkg/cloud"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+var ( // Declare exported dummy vars.
+	AffinityGroup      *cloud.AffinityGroup
+	CSCluster          *capcv1.CloudStackCluster
+	CAPIMachine        *capiv1.Machine
+	CSMachine1         *capcv1.CloudStackMachine
+	CAPICluster        *clusterv1.Cluster
+	CSMachineTemplate1 *capcv1.CloudStackMachineTemplate
+	Zone1              capcv1.Zone
+	Zone2              capcv1.Zone
+	Net1               capcv1.Network
+	Net2               capcv1.Network
+	ISONet1            capcv1.Network
+	Domain             string
+	DomainID           string
+	Account            string
+	Tags               map[string]string
+	Tag1               map[string]string
+	Tag2               map[string]string
+	Tag1Key            string
+	Tag1Val            string
+	Tag2Key            string
+	Tag2Val            string
+	CSApiVersion       string
+	CSClusterKind      string
+	CSClusterName      string
+	CSlusterNamespace  string
+	TestTags           map[string]string
+	CSClusterTagKey    string
+	CSClusterTagVal    string
+	CSClusterTag       map[string]string
+	CreatedByCapcKey   string
+	CreatedByCapcVal   string
+	LBRuleID           string
+	PublicIPID         string
+	EndPointHost       string
+	EndPointPort       int32
+)
+
+// SetDummyVars sets/resets all dummy vars.
+func SetDummyVars() {
+	// These need to be in order as they build upon eachother.
+	SetDummyCAPCClusterVars()
+	SetDummyCAPIClusterVars()
+	SetDummyCAPIMachineVars()
+	SetDummyCSMachineTemplateVars()
+	SetDummyCSMachineVars()
+	SetDummyTagVars()
+	LBRuleID = "FakeLBRuleID"
+}
+
+func CAPCNetToCSAPINet(net *capcv1.Network) *csapi.Network {
+	return &csapi.Network{
+		Name: net.Name,
+		Id:   net.ID,
+		Type: net.Type,
+	}
+}
+
+func CAPCZoneToCSAPIZone(net *capcv1.Zone) *csapi.Zone {
+	return &csapi.Zone{
+		Name: net.Name,
+		Id:   net.ID,
+	}
+}
+
+// SetDummyVars sets/resets tag related dummy vars.
+func SetDummyTagVars() {
+	CSClusterTagKey = "CAPC_cluster_" + string(CSCluster.ObjectMeta.UID)
+	CSClusterTagVal = "1"
+	CSClusterTag = map[string]string{CSClusterTagVal: CSClusterTagVal}
+	CreatedByCapcKey = "create_by_CAPC"
+	CreatedByCapcVal = ""
+	Tag1Key = "test_tag1"
+	Tag1Val = "arbitrary_value1"
+	Tag2Key = "test_tag2"
+	Tag2Val = "arbitrary_value2"
+	Tag1 = map[string]string{Tag2Key: Tag2Val}
+	Tag2 = map[string]string{Tag2Key: Tag2Val}
+	Tags = map[string]string{Tag1Key: Tag1Val, Tag2Key: Tag2Val}
+}
+
+// SetDummyClusterSpecVars resets the values in each of the exported CloudStackMachines related dummy variables.
```

Comment:
Note to self: Update comments for functions in this file. Many don't match the function name.